pyccel/codegen/printing/cwrappercode.py

```diff
@@ -273,88 +258,146 @@ def get_cast_function_call(self, cast_type, arg):
 
         return FunctionCall(cast_function, [arg])
 
+    # --------------------------------------------------------------------
+    # Functions that take care of collecting (data type, rank) checks and creating the error code
+    # --------------------------------------------------------------------
+    def _generate_TypeError_message(self, variable):
+        """
+        Generate TypeError message from the variable informations (datatype, precision)
+        """
+        dtype     = variable.dtype
+        precision = '' if isinstance(dtype, NativeBool) else '{} bit '.format(variable.precision * 2 * 8)\
+                       if isinstance(dtype, NativeComplex) else '{} bit '.format(variable.precision * 8)
+
+        message = '"{var_name} must be {precision}{dtype}"'.format(
+                        var_name  = variable,
+                        precision = precision,
+                        dtype     = self._print(variable.dtype))
+        return message
+
+    def _get_array_type_check(self, variable, collect_var):
+        """
+        Responsible for collecting array type check and creating the
+        corresponding error code
+        """
+
+        numpy_dtype = self.find_in_numpy_dtype_registry(variable)
+        arg_dtype   = self.find_in_dtype_registry(self._print(variable.dtype), variable.precision)
+
+        check = PyccelNe(FunctionCall(numpy_get_type, [collect_var]), numpy_dtype)
+        error = PyErr_SetString('PyExc_TypeError', '"{} must be {}"'.format(variable, arg_dtype))
+
+        return check, error
+
+    def _get_scalar_type_check(self, variable, collect_var, error_check = True):
+        """
+        Responsible for collecting numpy and python type check and creating the
+        corresponding error code
```

Comment:
Could you add `Parameters` and `Results` please

Reply:
done cfb3455